src/Microsoft.DotNet.Tools.Repl.Csi/Program.cs

```diff
@@ -14,26 +19,148 @@ public static int Main(string[] args)
         {
             DebugHelper.HandleDebugSwitch(ref args);
 
-            var app = new CommandLineApplication();
+            var app = new CommandLineApplication(throwOnUnexpectedArg: false);
             app.Name = "dotnet repl csi";
             app.FullName = "C# REPL";
             app.Description = "C# REPL for the .NET platform";
             app.HelpOption("-h|--help");
+
             var script = app.Argument("<SCRIPT>", "The .csx file to run. Defaults to interactive mode.");
+            var framework = app.Option("-f|--framework <FRAMEWORK>", "Compile a specific framework", CommandOptionType.MultipleValue);
+            var configuration = app.Option("-c|--configuration <CONFIGURATION>", "Configuration under which to build", CommandOptionType.SingleValue);
+            var preserveTemporary = app.Option("-t|--preserve-temporary", "Preserve the temporary directory containing the compiled project.", CommandOptionType.NoValue);
+            var project = app.Option("-p|--project <PROJECT>", "The path to the project to run. Can be a path to a project.json or a project directory", CommandOptionType.SingleValue);
 
-            app.OnExecute(() => Run(script.Value));
+            app.OnExecute(() => Run(script.Value, framework.Values, configuration.Value(), preserveTemporary.HasValue(), project.Value(), app.RemainingArguments));
             return app.Execute(args);
         }
 
-        private static int Run(string scriptOpt)
+        private static ProjectContext GetProjectContext(IEnumerable<string> targetFrameworks, string projectPath)
```

Comment:
A comment explaining the expected behavior (and a reference to whatever we used as a pattern for the behavior) here might be helpful.